builder/Dockerfile

```
@@ -1,4 +1,4 @@
-FROM golang:1.19.6-bullseye as builder
+FROM golang:1.23.1-bullseye as builder
```

As per the PR on @companieshouse/ci-golang-build#16 - Is there a reason why we're pinning to the minor version too? 
```suggestion
FROM golang:1.23-bullseye as builder
```

---

👍